src/components/NewRuntimeModal.js

```diff
@@ -139,16 +142,17 @@ export class NewRuntimeModalBase extends Component {
     const runtimeConfig = runtime?.runtimeConfig
     const gpuConfig = runtimeConfig?.gpuConfig
     const sparkMode = runtimeConfig?.cloudService === cloudServices.DATAPROC ? (runtimeConfig.numberOfWorkers === 0 ? 'master' : 'cluster') : false
+    const isDataproc = !sparkMode && !runtimeConfig?.diskSize
 
     return {
-      selectedPersistentDiskSize: disk?.size || DEFAULT_DISK_SIZE,
+      selectedPersistentDiskSize: disk?.size || DEFAULT_GCE_PERSISTENT_DISK_SIZE,
       sparkMode,
       masterMachineType: runtimeConfig?.masterMachineType || runtimeConfig?.machineType,
-      masterDiskSize: runtimeConfig?.masterDiskSize || runtimeConfig?.diskSize || DEFAULT_DISK_SIZE,
+      masterDiskSize: runtimeConfig?.masterDiskSize || runtimeConfig?.diskSize || (isDataproc ? DEFAULT_DATAPROC_DISK_SIZE : DEFAULT_GCE_BOOT_DISK_SIZE),
```

Comment:
hmm..this is a bit awkward. If it is gce, `masterDiskSize` means boot disk size? we don't intend to let user set bootDisk size, so we probably don't need to care for GCE (probably should just get rid of it in Leo)...

Reply:
Yea it is tough to follow too. `masterDiskSize` ends up being assigned the value of
-  `masterDiskSize` returned from Leo for Dataproc clusters
- `diskSize` returned from Leo for legacy GCE VMs without PDs

If we stop supporting GCE VMs without PDs, I think it can be simplified as

```javascript
masterDiskSize: runtimeConfig?.masterDiskSize || DEFAULT_DATAPROC_DISK_SIZE
```